boards/FSM/Src/App/App_AcceleratorPedals.c

```diff
@@ -171,3 +175,88 @@ float App_AcceleratorPedals_GetSecondaryPedalPercentage(const struct Accelerator
         accelerator_pedals->get_primary_encoder_counter_value(), accelerator_pedals->set_primary_encoder_counter);
         */
 }
+
+//TODO figure out what to do when primary/secondary is NAN, but signal has not activated yet.
+void App_AcceleratorPedals_Broadcast(struct FsmWorld* world)
+{
+    struct FsmCanTxInterface *can_tx = App_FsmWorld_GetCanTx(world);
+    struct AcceleratorPedals *accelerator_pedals = App_FsmWorld_GetPappsAndSapps(world);
+    struct Brake *            brake = App_FsmWorld_GetBrake(world);
+
+    const float papps_pedal_percentage = App_AcceleratorPedals_GetPrimaryPedalPercentage(accelerator_pedals);
+    const float sapps_pedal_percentage = App_AcceleratorPedals_GetSecondaryPedalPercentage(accelerator_pedals);
+
+    // update signals
+    const bool primary_pedal_alarm = papps_pedal_percentage == NAN;
+    const bool secondary_pedal_alarm = sapps_pedal_percentage == NAN;
+    SignalState papp_signal_state = App_SharedSignal_Update(
+        accelerator_pedals->papp_alarm_signal,
+        primary_pedal_alarm,!(primary_pedal_alarm || secondary_pedal_alarm));
+    SignalState sapp_signal_state = App_SharedSignal_Update(
+        accelerator_pedals->sapp_alarm_signal,
+        secondary_pedal_alarm, !(primary_pedal_alarm || secondary_pedal_alarm));
```

Comment:
Why does the exit condition require primary and secondary? Shouldn't primary only require primary not set, and likewise for secondary?

Reply:
@joethrstn is this a rule?

Reply:
ok fixed.